projects/step-frontend/src/lib/modules/execution/components/execution-legacy-switcher/execution-legacy-switcher.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, computed, effect, inject, input, Signal, signal } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Execution, ExecutionViewMode, ExecutionViewModeService } from '@exense/step-core';
+import { toSignal } from '@angular/core/rxjs-interop';
+import { map } from 'rxjs';
+
+@Component({
+  selector: 'step-execution-legacy-switcher',
+  templateUrl: './execution-legacy-switcher.component.html',
+  styleUrl: './execution-legacy-switcher.component.scss',
+})
+export class ExecutionLegacySwitcherComponent {
+  /** @Input() **/
+  execution = input<Execution | undefined>();
+
+  protected _executionViewModeService = inject(ExecutionViewModeService);
+
+  protected isDisabled = computed(() => {
+    const exec = this.execution();
+    return !exec || !this._executionViewModeService.isNewExecutionAvailable(exec);
+  });
+
+  protected toggleControl = new FormControl(false);
+
+  protected forceLegacyReporting = toSignal(this._executionViewModeService.checkForceLegacyReporting(), {
+    initialValue: false,
+  });
+
+  constructor() {
+    effect(() => {
+      if (this.execution()) {
+        this.updateToggleState();
+      }
+    });
+
+    this.toggleControl.valueChanges.subscribe((enabled) => {
+      this._executionViewModeService.setForceLegacyView(!enabled);
+      window.location.reload();
```

Comment:
Can you change the access to window, based on document injection?
```
private _doc = inject(DOCUMENT);

......

this._doc.defaultView?.location?.reload();
```

It's not critical, but it's recommend to access non angular objects through the injection also.
This might be useful, if in future the server-side rendering will be used. Accessing window object directly might rise an error in this case.